pkg/agent/openflow/pipeline.go

```diff
@@ -1216,16 +1245,20 @@ func (f *featurePodConnectivity) l3FwdFlowToPod(localGatewayMAC net.HardwareAddr
 			// This generates the flow to match the packets destined for a local Antrea IPAM Pod.
 			flows = append(flows, L3ForwardingTable.ofTable.BuildFlow(priorityNormal).
 				Cookie(cookieID).
+				MatchRegFieldWithValue(VLANIDField, uint32(vlanID)).
 				MatchProtocol(ipProtocol).
 				MatchDstIP(ip).
 				Action().SetDstMAC(podInterfaceMAC).
 				Action().GotoTable(L3DecTTLTable.GetID()).
 				Done())
 		} else {
 			// This generates the flow to match the packets with RewriteMACRegMark and destined for a local per-Node IPAM Pod.
-			flows = append(flows, L3ForwardingTable.ofTable.BuildFlow(priorityNormal).
-				Cookie(cookieID).
-				MatchProtocol(ipProtocol).
+			fb := L3ForwardingTable.ofTable.BuildFlow(priorityNormal).
+				Cookie(cookieID)
+			if f.connectUplinkToBridge {
+				fb = fb.MatchRegFieldWithValue(VLANIDField, 0)
+			}
```

Comment:
This is to block unexpected traffic. Only untagged traffic can go to NodeIPAM Pod.

Reply:
Perhaps add the comment to code, I had the question too.
And is it to "block" unexpected traffic? Will tagged packets towards a NodeIPAM Pod be dropped here? In my understanding, the condition is to avoid matching the traffic originated from AntreaIPAM Pod so it can go out and routed by physical network?

Reply:
Will tagged packets towards a NodeIPAM Pod be dropped here?
A: Yes, we don't support this traffic. A tagged packet must pass underlay GW and GW route this packet in proper VLAN.

In my understanding, the condition is to avoid matching the traffic originated from AntreaIPAM Pod so it can go out and routed by physical network?
A: Yes, this is implemented.

I think the accurate description is that "only overwrite MAC for untagged traffic which destination is regular NodeIPAM Pod".